### PR TITLE
Add some docs about working with prow/gcs/release controller data

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-403](https://issues.redhat.com//browse/TRT-403)

### CODE REVIEW

**Comment by stbenjam** [comment]
@xueqzhan Could you have a look?

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -6,38 +12,121 @@ Launch postgresql:
...
+  --database-dsn="postgresql://postgres:password@localhost:5432/postgres" \
+  --skip-bug-lookup
 ````
 
-### From a Prod Sippy Backup
+### From Prow and GCS buckets
 
-See the TRT team drive in Google, periodically there is a gzip'd backup stored there. Restore it locally with:
+Fetching data from prow directly gives us access to more raw data that TestGrid doesn't have, such as build cluster
+data, raw junit files, test durations, and more -- but it requires more configuration that using TestGrid.
```
Comment (xueqzhan) [inline DEVELOPMENT.md:70]: "than using TestGrid"

**Review by xueqzhan — COMMENTED** [review]
Lots of good information. Thanks

**Pull request merged**